app/javascript/components/concept-map/ConceptMap.css

```diff
@@ -59,7 +68,54 @@
 
     & .canvas {
         @apply absolute -z-1 top-0 left-0;
+        transition: opacity 120ms cubic-bezier(0.4, 0, 0.2, 1);
         touch-action: none;
-        /* border: 1px dashed; */
+        opacity: var(--c-concept-map-hover-opacity);
+
+        &.active {
+            opacity: 1;
+        }
+
+        &.hidden {
+            opacity: 0;
+        }
+
+        &.locked {
+            @apply -z-4;
+        }
+        &.available {
+            @apply -z-3;
+        }
+        &.complete {
+            @apply -z-2;
+        }
+
+        & path {
+            fill: none;
+        }
+
+        & .locked {
+            stroke: var(--c-concept-map-line-locked);
+            stroke-dasharray: var(--c-concept-map-line-dasharray-dashed);
+        }
+
+        & .available {
+            stroke: var(--c-concept-map-line-available);
+            stroke-dasharray: var(--c-concept-map-line-dasharray-dashed);
+        }
+
+        & .complete {
+            stroke: var(--c-concept-map-line-complete);
+            stroke-dasharray: var(--c-concept-map-line-dasharray-solid);
+        }
+
+        & .end-cap {
```

Comment:
```suggestion
        & circle {
```

Reply:
Rather than styling the element, I've used another named class to prevent potential cascade rule problems